CourseManager/Views/AddOrEditCourseSheet.swift

```diff
@@ -0,0 +1,141 @@
+
+    var body: some View {
+        NavigationView {
+            Form {
+                courseNameSection
+                courseURLSection
+                gitHubURLSection
+                statusPickerSection
+                if viewModel.course.status == .complete {
+                    dateCompletedCalendarSection
+                }
+                commentsSection
+                submitButtonSection
+                    .disabled(disableForm)
+            }.onAppear(perform: {
+                setUpViewModelCourseInformationToEdit()
+            })
+
+            .navigationTitle("Course Information")
+            .toolbar {
+                Button {
+                    handleXMarkTapped()
+                } label: {
+                    Image(systemName: "xmark")
+                        .font(.title2)
+                }
+            }
+        }
+    }
+
+    var courseNameSection: some View {
+        Section(header: Text("Course Name")) {
+            TextField("Enter course name", text: $viewModel.course.name)
+        }.textFieldStyle(.roundedBorder)
+    }
+
+    var courseURLSection: some View {
+        Section(header: Text("Course URL")) {
+            TextField("Enter course URL", text: $viewModel.course.url)
+        }.textFieldStyle(.roundedBorder)
+    }
+
+    var gitHubURLSection: some View {
+        Section(header: Text("GitHub URL")) {
+            TextField("Enter GitHub URL", text: $viewModel.course.gitHubURL)
+        }.textFieldStyle(.roundedBorder)
+    }
+
+    var statusPickerSection: some View {
+        Section {
+            Picker("Status", selection: $viewModel.course.status) {
+                ForEach(Status.allCases, id: \.self) { status in
+                    Text(status.localizedName)
+                        .tag(status)
+                }
+            }
+        }
+    }
+
+    var dateCompletedCalendarSection: some View {
+        Section(header: Text("Date completed")) {
+            DatePicker("Date completed", selection: $viewModel.course.dateOfCompletion, displayedComponents: .date)
+                .datePickerStyle(.compact)
+                .labelsHidden()
+        }
+    }
+
+    var commentsSection: some View {
+        Section(header: Text("Comments")) {
+            TextEditor(text: $viewModel.course.comments)
+        }
+    }
+
+    var submitButtonSection: some View {
+        Section {
+            Button {
+                handleSumitTapped()
+            } label: {
+                HStack {
+                    RoundedRectangle(cornerRadius: 10, style: .continuous)
+                        .overlay(alignment: .center) {
+                            Text("Submit")
+                                .fontWeight(.bold)
+                                .foregroundStyle(.white)
+                                .font(.title3)
+                        }
+                        .padding(.horizontal, 20)
+                }
+            }
+        }
+    }
+
+    var disableForm: Bool {
```

Comment:
`disableForm` is a good name for function, but for variable you want something like `isFormDisabled`

Reply:
updated